### PR TITLE
fix: entity object undefined in `afterUpdate` subscriber

### LINKED ISSUE #2809: Undefined entity - in subscribers after and before update events
**Issue type:**

[x ] bug report
**Database system/driver:**

[x] `sqlite`

**TypeORM version:**

[ x] `0.2.7 

**Steps to reproduce or a small repository showing the problem:**

Simple example here:  https://github.com/darekf77/typeorm-sqlite-error-example

### PR DESCRIPTION
**Pull request by francescoferraioli**
### Description of change

Similar to #5443, just for the `afterUpdate` subscribers.

This, alongside #5443 should fix the bug raised in https://github.com/typeorm/typeorm/issues/2809 and https://github.com/typeorm/typeorm/issues/2246.

### Pull-Request Checklist

- [x] Code is up-to-date with the `master` branch
- [x] `npm run lint` passes with this change
- [x] `npm run test` passes with this change
- [x] This pull request links relevant issues as `Fixes #2809`
- [x] There are new or updated unit tests validating the change
- [x] Documentation has been updated to reflect this change
- [x] The new commits follow conventions explained in [CONTRIBUTING.md](https://github.com/typeorm/typeorm/blob/master/CONTRIBUTING.md)

### CODE REVIEW

**Review by imnotjames — APPROVED** [review]
This makes sense to me and the tests look good.

**Review by imnotjames — CHANGES_REQUESTED** [review]

File: test/github-issues/2809/subscriber/PostSubscriber.ts [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+
+@EventSubscriber()
+export class PostSubscriber implements EntitySubscriberInterface<Post> {
+    listenTo() {
+        return Post;
+    }
+
+    afterUpdate(event: UpdateEvent<Post>) {
+        if (event.entity) {
+            event.entity["title"] = "set in subscriber after updated";
```
[the remark this answers is not in the document]
Comment (francescoferraioli) [reply test/github-issues/2809/subscriber/PostSubscriber.ts:12]: in an update you can pass in a partial entity, only the properties you want updated.

[the remark this answers is not in the document]
Comment (imnotjames) [reply test/github-issues/2809/subscriber/PostSubscriber.ts:12]: I understand.  However, the definition of `UpdateEvent` REQUIRES the actual entity.  Not the partial.

[the remark this answers is not in the document]
Comment (francescoferraioli) [reply test/github-issues/2809/subscriber/PostSubscriber.ts:12]: Yeah I agree that the type is not strictly correct. I think the problem is that the `UpdateEvent` is called both when you use `save` and when you use `update`. In the case of `save` it's the full entity but in the case of `update` it's a partial.

The same could be said about the existing `beforeUpdate` subscriber as that is currently being passed a partial entity but the type doesn't reflect that.

As we currently are doing the exact same thing for `beforeUpdate`, I think it's safe to do it for `afterUpdate` and fixing the type could be done separately.

[the remark this answers is not in the document]
Comment (imnotjames) [reply test/github-issues/2809/subscriber/PostSubscriber.ts:12]: If that's the case, can you update the types to match that?

[the remark this answers is not in the document]
Comment (francescoferraioli) [reply test/github-issues/2809/subscriber/PostSubscriber.ts:12]: I have corrected the type at https://github.com/typeorm/typeorm/pull/7783

[the remark this answers is not in the document]
Comment (francescoferraioli) [reply test/github-issues/2809/subscriber/PostSubscriber.ts:12]: I have done it as a seperate PR for a number of reasons:
- The type fixing spans more than just the `afterUpdate`
- It is a breaking change whilst this is not

I think PRs should be very targeted at fixing one thing at a time :)

[the remark this answers is not in the document]
Comment (francescoferraioli) [reply test/github-issues/2809/subscriber/PostSubscriber.ts:12]: @imnotjames - is there any further reason not to accept this PR?

**Comment by francescoferraioli** [comment]
@imnotjames - is there any further reason not to accept this PR?

File: test/github-issues/2809/subscriber/PostSubscriber.ts (hunk 1 above)
[the remark this answers is not in the document]
Comment (francescoferraioli) [reply test/github-issues/2809/subscriber/PostSubscriber.ts:12]: @imnotjames - now that https://github.com/typeorm/typeorm/pull/7783 is in, can we please also accept and merge this one?

**Review by imnotjames — APPROVED** [review]

**Pull request merged**